2023 Pricing Grid: Use smallest unit for currencies in the plans pricing grid

jeyip opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/pull/81338#discussion_r1318929055

## Context

With the new storage add-ons feature, a user can add storage upgrades when they purchase a plan in onboarding. What this means is that there's a point where, behind the scenes, we need to calculate add-on prices + plan prices in a dynamic way. The arithmetic operation would ideally look something like $20 add-on + $20 plan = $40 total. There is, however, a hiccup.

Right now, `add-on` prices only return in the smallest unit ( Ex. USD $20 is represented in a value that corresponds to cents, so it's returned as the integer 2000 ). This is what's recommended by the payments team so as to avoid rounding errors from floating point calculations.

`plan` prices, however, are not handled in the smallest unit. ( Ex. USD $20 is represented as dollars, so it's handled as the integer 20 )

Fixing this representation mismatch is what we're trying to address in this PR. We are changing all plan prices to use smallest units for a given currency.

## Proposed Changes

* ~Ensure changes only target the `/start` flow~ from @southp: it's no longer the case :) 
* Changes all prices in the 2023 plan features grid to use the smallest unit a currency provides

## Screenshots
### Plans Features Grid
<img width="1477" alt="Screenshot 2023-09-12 at 5 56 06 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/f0fe60b0-2228-4b0b-9523-5d0c54eb1dec">

### Plans Comparison Grid
<img width="1326" alt="Screenshot 2023-09-12 at 5 56 14 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/ebf87221-9725-46eb-9c47-60c0fee7a327">

### Sticky CTA Buttons
<img width="1474" alt="Screenshot 2023-09-12 at 6 23 35 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/8db852ce-7747-43d9-8880-c938d086bcb2">

## Testing Instructions
<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Check out this PR
* Navigate to `/start/plans`
* Verify that plans in the pricing headers and billing timeframes display as expected in the plans comparison grid and the plans features grid.
* Verify that prices display correctly in the sticky CTA buttons.
* Repeat verification with different WordPress currencies if possible USD, EUR, AUD, GBP, BRL, CAD, CHF, CZK, DKK, HKD, HUF, IDR, ILS, INR, JPY, MXN, NOK, NZD, PHP, PLN, RUB, SEK, SGD, THB, TWD, TRY
* Repeat verification with different term prices if available ( monthly vs. annually vs. 2yearly vs. 3yearly )
* Repeat verification in the following flows PCYsg-RDT-p2
* Repeat verification for the Calypso admin `/plans` page ( wordpress.com/plans/{SITE_SLUG} )

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

jeyip committed:
Use smallest unit for currencies in the plans pricing grid

jeyip commented on client/my-sites/plans-features-main/index.tsx:
```diff
@@ -239,6 +239,7 @@ const PlansFeaturesMain = ( {
 	const { setShowDomainUpsellDialog } = useDispatch( WpcomPlansUI.store );
 	const domainFromHomeUpsellFlow = useSelector( getDomainFromHomeUpsellInQuery );
 	const showUpgradeableStorage = config.isEnabled( 'plans/upgradeable-storage' );
+	const shouldUseSmallestUnitCurrency = flowName === 'onboarding';
```
**Note:**

We'd like to scope the smallestUnit currency changes to the `/start` flow for minimal impact. If other folks have suggestions about better ways to limit the surface area of the changes, feel free to chime in 🙂

jeyip commented:
**Note:**
Currently fixing typescript errors and doing a second pass of the code that I've written 👁️

jeyip committed:
Fix type issues in plans mocks

jeyip committed:
Simply getPlanRawPrice selector

jeyip committed:
Reformat inline comments about returnSmallestUnit

jeyip committed:
Refactor getDiscountedRawPrice selector

jeyip committed:
Refactor getPlanDiscountedRawPrice

jeyip committed:
Refactor get site plan raw price

jeyip commented on client/my-sites/plan-features-2023-grid/hooks/npm-ready/use-is-large-currency.ts:
```diff
@@ -15,12 +16,18 @@ interface Props {
 			discountedPrice[ returnMonthly ? 'monthly' : 'full' ],
-		].some( ( price ) => price && price.toString().length > LARGE_CURRENCY_CHAR_THRESHOLD );
+		].some( ( price ) => {
+			const formattedPrice = formatCurrency( price || 0, currencyCode || '', {
```
**Note:**

@sirbrillig This hook determines if the number of numerical digits for a display price are too long to represent properly in the plans pricing grid. Since we’re trying to switch over to use smallest units, do you have happen to have suggestions on how to best account for this ( since $20 is now represented as the integer 2000, instead of 20 )? I thought about using `formatCurrency` or `getCurrencyObject` to account for smallest units, but the return values from those functions are display strings that include currency symbols and place separators ( periods, commas, etc. ).

I’m wondering if this is a valid use case to modify the interface of the `format-currency` library to provide the necessary info. Alternatively, I might be missing something, so I’d love your thoughts

sirbrillig replied:
🤔 I have two questions:

1. Is there any way to use CSS for this sort of display logic rather than code? For example, it looks like [here](https://github.com/Automattic/wp-calypso/blob/4d8d10c2b6ae92e0fec9d2ec679bcad9f326dc3c/client/my-sites/plan-features-2023-grid/components/actions.tsx#L88-L103) a label without the price is displayed if the currency is "large". Another option might be to render both the "with price" and "without price" versions and hide one using a [css container query](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_container_queries) unless the container is large enough to display the price? 
2. If the concern is sufficient space to display the formatted price, don't you want to know the length of the price as a string with formatting? The price `100.50 HK$` is 5 numbers (6 if you count the decimal) but takes up 10 characters, whereas the price `¥505001` also has 6 numbers but takes up only 7 characters. Some currencies have relatively long currency symbols like GEL which uses `Lari` or VEF which uses `Bs. F.`. 

Also worth noting that not all characters are the same width (especially in some languages), which makes using a "number of digits" for display fit purposes somewhat dubious.

jeyip replied:
Thanks for the insight Payton! 

> Is there any way to use CSS for this sort of display logic rather than code? For example, it looks like [here](https://github.com/Automattic/wp-calypso/blob/4d8d10c2b6ae92e0fec9d2ec679bcad9f326dc3c/client/my-sites/plan-features-2023-grid/components/actions.tsx#L88-L103) a label without the price is displayed if the currency is "large". Another option might be to render both the "with price" and "without price" versions and hide one using a [css container query](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_container_queries) unless the container is large enough to display the price?

@aneeshd16, I believe you worked on this? I'm wondering if these approaches were considered and if there's any hidden gotchas to using CSS containers?

> If the concern is sufficient space to display the formatted price, don't you want to know the length of the price as a string with formatting? The price 100.50 HK$ is 5 numbers (6 if you count the decimal) but takes up 10 characters, whereas the price ¥505001 also has 6 numbers but takes up only 7 characters. Some currencies have relatively long currency symbols like GEL which uses Lari or VEF which uses Bs. F..

Your point about relatively long currency symbols and your example with HK$ makes sense to me. 

My best guess is that perhaps the smaller formatting and font-sizes for currency symbols and separator symbols might make their widths relatively negligible, although again, with examples like GEL, I do wonder if there are overflow issues somewhere. @aneeshd16 wondering if you have thoughts here as well?

I did a very non-scientific comparison:

<img width="237" alt="Screenshot 2023-09-12 at 10 22 49 AM" src="https://github.com/Automattic/wp-calypso/assets/5414230/bc7cf69f-eff1-4527-a0d0-b67bf9ac2447">

From what I can tell right now, the current implementation isn't ideal. Counting numerical digits does, however, seem to work as a proxy given our existing currency support ( the currency ends up being low even if the symbol is large, and vice versa ).

I'm going to check different currencies, but since these changes are already in production though @sirbrillig + @aneeshd16, I'm wondering how we'd feel about adjusting the `LARGE_CURRENCY_THRESHOLD` for now and exploring these ideas in a follow-up, with the consideration that this is relatively higher priority given that new currencies could potentially break formatting in the future?

_Edit:_
I've checked every single currency we support from USD to TRY. Everything looks reasonable if we adjust the threshold from 5 to 6 -- happy to provide screenshots

sirbrillig replied:
If you don't care about the currency symbol, one other idea as an interim fix might be to use `getCurrencyObject()`:

```js
const currencyObject = getCurrencyObject( /* ... */ );
const priceToMeasure = currencyObject.integer + currencyObject.hasNonZeroFraction ? currencyObject.fraction : '';`
const priceLength = priceToMeasure.length;
const isPriceTooLong = priceLength > LARGE_CURRENCY_THRESHOLD;
```

That way you are including all the other price formatting (commas, spaces, and periods) in the measurement.

jeyip replied:
Made the interim fix in https://github.com/Automattic/wp-calypso/pull/81537/commits/ba8aa5b5d8ed11f2a706452ef6e50419fdedf8b0

aneeshd16 replied:
Hi, thanks for the ping!

> I believe you worked on this? I'm wondering if these approaches were considered and if there's any hidden gotchas to using CSS containers?

The challenge was that all prices in the pricing grid should be displayed in the same font size. There is a possibility that the price of one plan exceeds the threshold while others don't. For example, in certain currencies, the price of the eCommerce plan may be classified as "large" based on our heuristic, while the price of the free plan (0) will not be "large". This is why we cannot use CSS solutions that act on a single element, as we need to consider all currencies displayed on the grid. This is also why this hook accepts a list of plans instead of a single plan.

Since we need to consider currency symbols and space/decimal/comma/etc characters as well, why don't we check for the length of the string produced by `formatCurrency`? This would have the same result as the current implementation.

```js
const formattedPrice = formatCurrency( price || 0, currencyCode || '', {
  stripZeros: true,
  isSmallestUnit: true,
} );
return formattedPrice.length > LARGE_CURRENCY_CHAR_THRESHOLD;
```

> I've checked every single currency we support from USD to TRY. Everything looks reasonable if we adjust the threshold from 5 to 6 -- happy to provide screenshots

I don't see anything wrong with this approach either since `LARGE_CURRENCY_CHAR_THRESHOLD` was also based on observation and not on any other rule. 👍

jeyip committed:
Fix useIsLargeCurrency hook

jeyip committed:
Use smallest units for all pricing in 2023 plans pricing grid

jeyip commented:
Currently:

Fixing unit tests 👁️

jeyip commented:
**Note:**

The way this PR was originally designed,https://github.com/Automattic/wp-calypso/pull/81537#discussion_r1319279648. @southp pointed out that the code complexity to make this happen wasn't ideal, which I agree with.

I _do_ like the safety of limiting the surface area of our changes, but it would also leave the 2023 plans pricing grid in a state where we were conditionally using smallest units for price display. In other words, it would be another opportunity for bugs to creep into how prices are rendered, which, given how important the topic is to both us and our users, isn't worth the tradeoff.

I recognize that exhaustively testing pricing can be a challenging process, but we've had soft commitment from @karenroldan to assist with this. If anyone has thoughts, I'd be happy to listen to other suggestions about how we could better limit these changes so that they could be released in a more controlled manner.

jeyip committed:
Fix tests

southp commented on client/state/sites/plans/selectors/get-plan-discounted-raw-price.ts:
```diff
@@ -12,23 +12,27 @@ export function getPlanDiscountedRawPrice(
+	const rawPrice = plan?.rawPrice ?? -1;
 
-	if ( ( plan?.rawPrice ?? -1 ) < 0 || ! isSitePlanDiscounted( state, siteId, productSlug ) ) {
+	if ( rawPrice < 0 || ! isSitePlanDiscounted( state, siteId, productSlug ) || ! plan ) {
```
Nit: by rawPrice < 0, it actually imples `! plan` from how rawPrice is derived. We can also move `if ( ! plan ) { return null; }` right above `rawPrice` since we know the subsequent computation isn't needed if `plan` is falsey. It's very minor so please keep it for now :)

southp reviewed (approved):
I've checked the following currencies in `/start`, `/plans`, and `/setup/newsletter`:

* USD
* TWD
* EUR
* INR
* PHP (for the one-time discount)
* TRY
* IDR (for the longer currency)

Everything works as expected to me. I've also manually triggered the Pre-release test and [it has passed](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Calypso_E2E_Pre_Release?mode=builds&hideProblemsFromDependencies=false&hideTestsFromDependencies=false#10777945).

The PR even manages to fix one production bug where no price is shown in the Newsletter flow for the Free plan for certain currencies.

| Production | this PR |
| ----------- | -------|
| ![image](https://github.com/Automattic/wp-calypso/assets/1842898/0aac7901-b218-4c6a-ad78-6b2c770939d3) | ![image](https://github.com/Automattic/wp-calypso/assets/1842898/39302fa6-e407-41b6-87e8-dfd2eefdced9) |

and the improved `isLargeCurrency()` has improved the space utilization in some cases, e.g. in TRY:
| Production | this PR |
| ----------- | -------|
|![image](https://github.com/Automattic/wp-calypso/assets/1842898/4ceafff5-f2c6-441a-9b77-c5d37daa4e43)| ![image](https://github.com/Automattic/wp-calypso/assets/1842898/12739628-dd4d-47ef-9ea1-e9d4c3907b90)|

I also feel your solution is better than my proposal, @jeyip . Thanks for the craftsmanship of identifying an even better way :) 

I've approved it, but let's still wait for a day for KitKat to chime in for the maximum safety :)

karenroldan commented:
Hi all! I'm summarizing my findings below from today's testing. The currencies I focused on are **USD**, **SEK**, **HUF**, **IDR**, and **JPY** because they offer a combination of typical currencies and those with unique smallest unit characteristics.

The flows I tested based on PCYsg-RDT-p2 are the following:
- /plans/[paid | free]
- /setup/design-first
- /setup/domain-upsell
- /setup/link-in-bio
- /start
- /start/domains

I'll be grouping them based on the issues I found:

❌  **UI issue in wordpress.com/plans/{SITE_SLUG}:** If there’s an upgrade credit (in my case it’s $50) and the discounted price includes decimal values, the text becomes too wide for its designated space. The original price overflows into the next column.

**USD** 

![CXld6A.png](https://github.com/Automattic/wp-calypso/assets/67279475/1de98bce-fb50-4489-a332-5cd2a4dc7368)

| JPY - Prod  | JPY - This PR |
| ------------- | ------------- |
| ![eHq7Pf.png](https://github.com/Automattic/wp-calypso/assets/67279475/2dbe6beb-7588-4442-97c8-586ebdac1e65)  | ![u5zACR.png](https://github.com/Automattic/wp-calypso/assets/67279475/4d2981c6-2e00-49ff-9896-94e27368ad53)  |
| ![AFBmEN.png](https://github.com/Automattic/wp-calypso/assets/67279475/ae40a267-a057-453f-bfc1-6196963e35d0)  | ![MJWliq.png](https://github.com/Automattic/wp-calypso/assets/67279475/3cbf1f95-df04-4d30-845f-a89123a730f6)  |

❌  **No price is shown in the /setup/domain-upsell/plans for the Free plan.** The issue is also in prod but wanted to note it here since the changes fixed the same issue in the [/newsletter](https://github.com/Automattic/wp-calypso/pull/81537#pullrequestreview-1623573144) flow. I checked with USD, HUF, JPY.: 

_Note_: This PR also fixed the missing price in /setup/link-in-bio.

![1B28DS.png](https://github.com/Automattic/wp-calypso/assets/67279475/8684b160-1ccf-45d7-81bb-c5258c03283e)

❌ ❌ **Prices are not rendering correctly in IDR:** I checked in `/plans`, `/start`, and `/link-in-bio` and observed different behaviors: 

In the `/plans` page, the prices displayed are incorrect for both Business and Commerce plans. 

| Plan  | Price in Prod | Price in this PR | 
| ------------- | ------------- | ------------- | 
| Business  | Rp 298,000  | Rp 2,980  | 
| Commerce  | Rp 484,333.33  | Rp 4,843.33  | 
| Screenshot  | ![HDP5sJ.png](https://github.com/Automattic/wp-calypso/assets/67279475/163955cb-07f5-4465-851d-be6c2f27839a)  | ![EupHtL.png](https://github.com/Automattic/wp-calypso/assets/67279475/bb890116-34be-4b3b-b9b6-11372c18aed8)  |

I shared a video [here](p1694611934692969/1694477120.445689-slack-CQD1HH4MA) that shows the total of Rp 4,843.33 on the checkout page.

In that video, you will notice that when I didn't proceed with the payment and went back to the previous page, the correct prices were displayed. 

It also seems that the page is rendering differently every time you visit it. This time the price for the Personal plan went down from Rp 44,700 to Rp 447. Similar to the Premium plan, it went down from Rp 89,400 to Rp894.

![0D6Dfs.png](https://github.com/Automattic/wp-calypso/assets/67279475/3b736224-cbc5-4ca4-8498-3d82d63bfcc2)

I also tested on `/link-in-bio` flow and the prices for IDR are incredibly low: 

![zKvT6T.png](https://github.com/Automattic/wp-calypso/assets/67279475/da81c2a9-53b7-4d05-bbeb-21f749c1fb79)

I've also observed that the total prices on the checkout page vary: sometimes it displays the correct values, and sometimes it displays the lower values.

jeyip committed:
Fix currency overflow issues

Previously, useIsLargeCurrency was only evaluating  prices individually. It, however, wasn't considering when discounted and undiscounted prices were displayed on the same line. Because of this, we were seeing overflow in the plans comparison grid for AUD, CAD, CHF, etc.

To remedy this, we set a threshold and now evaluate the combined discounted and undiscounted prices for a given plan to decide whether or not a currency is large for its given display

To be clear, this is not an ideal solution. We should explore CSS containers for a more exact way to handle price formatting

sirbrillig commented on client/my-sites/plan-features-2023-grid/hooks/npm-ready/use-is-large-currency.ts:
```diff
@@ -2,15 +2,36 @@ import { getCurrencyObject } from '@automattic/format-currency';
 	returnMonthly?: boolean;
 }
 
+function formatPrice( price: number, currencyCode: string ) {
```
FWIW, this function is now nearly exactly what `formatCurrency()` does, except that it doesn't take into account which side of the price the currency symbol is on (not that it probably matters in this context). You could probably just use `formatCurrency()` instead.

jeyip replied:
Thanks for the heads up! I'll take a look in a few moments 👀

jeyip replied:
Made the change in https://github.com/Automattic/wp-calypso/pull/81537/commits/82dd692cc4c549491e051fdbdd99f50b5dd7da30 😎

jeyip commented:
Hi @karenroldan -- wonderful testing 🙏 

> ❌ UI issue in wordpress.com/plans/{SITE_SLUG}: If there’s an upgrade credit (in my case it’s $50) and the discounted price includes decimal values, the text becomes too wide for its designated space. The original price overflows into the next column.

I believe I've fixed this in the scenarios you're describing. In doing so, I believe I've also fixed price overflow in the plans comparison grid for AUD,CAD,CHF,DKK,HUF,NZD,PLN,RUB,SGD currencies.

**Note:**
I'll edit this comment as I address the other testing issues you've mentioned

southp commented:
@jeyip @karenroldan I think the IDR issue is likely due to the prices are defined differently in production vs in sandbox, since I can see that price difference simply by sandboxing / unsandboxing:

https://github.com/Automattic/wp-calypso/assets/1842898/ab6c61c5-a293-4fcf-bcf0-d6f2e9bed470

@sirbrillig Can you help us confirm? 

If that's the case, once we agree on the discount layout fix, this PR will be good to go.

southp commented:
* [The discount label fix](https://github.com/Automattic/wp-calypso/pull/81537/commits/82dd692cc4c549491e051fdbdd99f50b5dd7da30) LGTM
* Let's not address the `/setup/domain-upsell/plans` issue for now to not further bloat this PR.

southp commented:
Confirmed that the `orig_cost_integer` is two trailing zeroes different.
 
Sandboxed:
<img width="1023" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1842898/55503031-2a93-4472-b10b-0462eaea8664">

Production:
<img width="1135" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1842898/b7bbba3d-d291-4d72-9df6-5234c2c5584c">

In that case, I think this PR should be safe to go into production. @Automattic/shilling could you help us confirm? I'd be grateful.

sirbrillig commented:
🤔 That's interesting. As far as I can tell the Premium plan price in IDR on the test tables is the same as the production tables. I haven't been able to reproduce the price with the extra digits. I always get `1072800`, or `Rp 1,072,800` (since IDR is a zero decimal currency), which is the plan's current price from what I can tell.

I see from your screenshots that all the integer prices have the extra zeroes, but the formatted `price` does not. The additional zeros suggests that PHP thinks IDR has decimal places, if that was the case, `107280000` would indeed be formatted as `Rp 1,072,800`, which is what we see.

Something I have noticed is that sometimes when formatting IDR it shows up as `Rp 1,072,800` and sometimes `Rp1,072,800` (no space). I assumed this was just a fluke, but I wonder if it's related to this issue. Both the formatting as well as the number of decimals comes [from PHP itself](https://www.php.net/manual/en/numberformatter.formatcurrency.php). I wonder if there's an issue where some of our servers have a different version of that library. I made https://github.com/Automattic/payments-shilling/issues/2022 to investigate. I don't think it has anything to do with this PR, but if you could help me reproduce the bug I'd appreciate it.

karenroldan commented:
> I can't quite seem to replicate the issues you were seeing. It appears as if it might have to do with caching, especially with prices jumping between values as you visit other pages. Would you be able to describe your testing workflow?
> 
> * Are you sandboxing any environment ( Ex. public api, wordpress.com, etc. )?
> * Are you using the local dev environment or calypso.live?
> * Are you switching between currencies relatively quickly?
> * Are you stepping through an entire flow from onboarding to the calypso admin /plans page?
> 
> Here's what I tried to do to reproduce:
> 
> * No sandboxes
> * switch from USD to IDR currency
> * navigate to calypso.localhost:3000/start
> * pick either a free plan or a paid plan and finished onboarding
> * navigated to `/plans` in calypso admin
> * repeated the same steps with container-admiring-joliot.calypso.live/start ( calypso.live )
> 
> If this isn't a caching issue, the problems that you're seeing indicate that incorrect values are being returned from the `/plans` and the `/plans/site` API endpoint 🤔 I'm going to mull this over more, because I'm unsure what the cause of this is.

Hi @jeyip and @southp thank you for the detailed investigation! I left my account yesterday to continue using the IDR currency and did a round of testing today. 

1. I'm not sandboxing any environment. 
2. I am using the calypso.live link 
3. I thought switching between currencies quickly might be the cause. I tried it again and saw the same values. 😕 
4. I tried two scenarios here: (1) starting from `/start` from onboarding to the Calypso /plans page and (2) from my existing site, I navigated to the /plans page. 

Sharing the video of today's recording: 

1. No sandboxes 
2. IDR currency has been set on my account since yesterday

**Starting from /start, choose a paid plan, and navigate to the checkout page.** Observations: 

1. The prices displayed in `/start/plans/` are the expected values. 
2. Selecting the Commerce plan (annual) gives me a total of Rp 70,415 only. This includes my credit and the 25% VAT.  
3. I did not continue with the payment and got the Free plan. 
4. In the /plans, I now see the lowered values. 
5. When I tried upgrading to the Business plan, I saw from the skeleton screen that the correct prices were initially displayed and finally, the total became Rp 35,015. 

https://github.com/Automattic/wp-calypso/assets/67279475/682266a2-c7a3-4acd-acc5-4a49cffeb0f0

karenroldan commented:
✅  Confirming that this issue is already fixed. 🥳 

> ❌ UI issue in wordpress.com/plans/{SITE_SLUG}: If there’s an upgrade credit (in my case it’s $50) and the discounted price includes decimal values, the text becomes too wide for its designated space. The original price overflows into the next column.

| Currency  | Desktop | Mobile |
| ------------- | ------------- | ------------- |
| USD  | ![JvzrJG.png](https://github.com/Automattic/wp-calypso/assets/67279475/b5a9117b-ffb0-4bcb-8fed-15f59e22b5ca)  | ![zexmSq.png](https://github.com/Automattic/wp-calypso/assets/67279475/a87be717-d5e6-4f88-bf75-ca9d4fe261f4)  |
| JPY  | ![ElZNDf.png](https://github.com/Automattic/wp-calypso/assets/67279475/458a532f-1f8f-4d8d-aafa-98400087c34d)  |![HEKhtz.png](https://github.com/Automattic/wp-calypso/assets/67279475/0a937253-326c-45ce-81c7-abf514613499)  |
